woocommerce/woocommerce
# E2E tests migration: performance -> cart

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42068

With this PR, we're migrating the `performance -> cart` tests from Puppeteer to Playwright. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Navigate to `/plugins/woocommerce-blocks` and run `pnpm env:restart` to set up your environment.
3. Run `pnpm playwright test --config=tests/e2e/playwright.side-effects.config.ts -- plugins/woocommerce-blocks/tests/e2e/tests/cart/cart-block.performace.block_theme.side_effects.spec.ts` or use code editor UI to run the tests. 
4. Ensure the Cart performance tests are passing. 
5. Verify the performance report present at `e2e/utils/performance/reports/e2e-cart-performance.json` looks similar to below:
```
{"description":"Cart block loading: (serverResponse)","longest":4237,"shortest":2735,"average":3250.6666666666665}
{"description":"Cart block loading: (firstPaint)","longest":673,"shortest":416,"average":505}
{"description":"Cart block loading: (domContentLoaded)","longest":3907,"shortest":2936,"average":3264.3333333333335}
{"description":"Cart block loading: (loaded)","longest":3912,"shortest":2942,"average":3270}
{"description":"Cart block loading: (firstContentfulPaint)","longest":673,"shortest":416,"average":505}
{"description":"Cart block loading: (firstBlock)","longest":4473,"shortest":3072,"average":3549}
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Migrate `performance -> cart` tests from Puppeteer to Playwright. 

#### Comment 

</details>
